server/admin_management/api/app/services/access_control_privilege_service.py

```diff
@@ -0,0 +1,150 @@
+import logging
+from http import HTTPStatus
+from sqlalchemy.orm import Session
+from typing import List
+
+from api.app import constants as famConstants
+from api.app import schemas
+
+from api.app.services.user_service import UserService
+from api.app.services.role_service import RoleService
+from api.app.repositories.access_control_privilege_repository import (
+    AccessControlPrivilegeRepository,
+)
+
+from api.app.utils import utils
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AccessControlPrivilegeService:
+    def __init__(self, db: Session):
+        self.user_service = UserService(db)
+        self.role_service = RoleService(db)
+        self.access_control_privilege_repository = AccessControlPrivilegeRepository(db)
+
+    def get_acp_by_user_id_and_role_id(self, user_id: int, role_id: int):
+        return self.access_control_privilege_repository.get_acp_by_user_id_and_role_id(
+            user_id, role_id
+        )
+
+    def get_acp_by_id(self, access_control_privilege_id: int):
+        return self.access_control_privilege_repository.get_acp_by_id(
+            access_control_privilege_id
+        )
+
+    def create_access_control_privilege(
+        self, request: schemas.FamAccessControlPrivilegeCreateRequest, requester: str
+    ) -> List[schemas.FamAccessControlPrivilegeCreateResponse]:
+        LOGGER.debug(
+            f"Request for assigning access role privilege to a user: {request}."
+        )
+
+        # Verify if user already exists or add a new user
+        fam_user = self.user_service.find_or_create(
+            request.user_type_code, request.user_name, requester
+        )
+
+        # Verify if role exists
+        fam_role = self.role_service.get_role_by_id(request.role_id)
+        if not fam_role:
+            error_msg = f"Role id {request.role_id} does not exist."
+            utils.raise_http_exception(HTTPStatus.BAD_REQUEST, error_msg)
+
+        # Role is a 'Abstract' type, create access control privilege with forst client child role
+        require_child_role = (
+            fam_role.role_type_code == famConstants.RoleType.ROLE_TYPE_ABSTRACT
+        )
+
+        create_return_list: List[schemas.FamAccessControlPrivilegeCreateResponse] = []
+
+        if require_child_role:
+            LOGGER.debug(
+                f"Role {fam_role.role_name} requires child role "
+                "for creating delegate admin access control privilege."
+            )
+
+            if (
+                not hasattr(request, "forest_client_number")
+                or request.forest_client_number is None
+            ):
+                error_msg = (
+                    "Invalid access control privilege request, missing forest client number."
+                )
+                utils.raise_http_exception(HTTPStatus.BAD_REQUEST, error_msg)
+
+            for forest_number in request.forest_client_number:
+                # Check if child role exists or add a new child role
```

Comment:
We suppose to validate the forest client number here. But there are too much code in this ticket, created a follow up ticket #1139 to add the validation.